### PR TITLE
Refactor force basic auth, now all modules which use fetch_url() can use force_basic_auth

### PR DESCRIPTION
**Pull request by woltage**
Check out PR: https://github.com/ansible/ansible/pull/9281 which is related to this PR.

### CODE REVIEW

**Comment by gregdek** [comment]
@woltage thanks for submitting this pull request. Please accept our apologies for taking so long to review it.

Unfortunately, due to this delay, it appears that the underlying code has changed, and this PR needs to be rebased. If you believe that this PR still applies, please rebase and we will then put it into our review process as soon as possible.  Again, apologies for the delay and the extra work.

**Comment by woltage** [comment]
Fixed, but I need to fix ansible/ansible#9281 too.

**Comment by gregdek** [comment]
@jpmens can you review? If you give this a +1 we will tag "P1_reviewed" for quicker inclusion.

**Comment by jpmens** [comment]
+1

File: network/basics/get_url.py [hunk 1]
```diff
@@ -108,6 +108,15 @@
         parameter is not specified, the C(url_password) parameter will not be used.
     required: false
     version_added: '1.6'
+  force_basic_auth:
```
Comment (bcoca) [inline network/basics/get_url.py]: requires version added

File: network/basics/uri.py [hunk 2]
```diff
@@ -369,7 +368,6 @@ def main():
             body_format = dict(required=False, default='raw', choices=['raw', 'json']),
             method = dict(required=False, default='GET', choices=['GET', 'POST', 'PUT', 'HEAD', 'DELETE', 'OPTIONS', 'PATCH']),
             return_content = dict(required=False, default='no', type='bool'),
```
Comment (bcoca) [inline network/basics/uri.py:371]: patch seems to remove feature, not add it

Comment (jpmens) [inline network/basics/uri.py:371]: I thought so too until I looked at the linked issue which seems to add it there.

**Comment by bcoca** [comment]
ah, this one does remove it, but the local version and the other request makes it global.

**Comment by bcoca** [comment]
waiting for rebase in the other ticket, won't remove the feature till it is added there

**Comment by woltage** [comment]
@jpmens @bcoca: My mistake look's like uri -module is not using module_utils/urls.py from Ansible core. I think it would be nice rewrite uri -module to use urls.py. (let's see if will have some time on my vacation)

**Comment by woltage** [comment]
@bcoca: It would be great if you could merge this too.  You already merged ansible/ansible/pull/9281

**Pull request merged**